src/main/java/de/tum/in/www1/artemis/service/connectors/localci/LocalCISharedBuildJobQueueService.java

```diff
@@ -51,9 +51,16 @@ public class LocalCISharedBuildJobQueueService {
 
     private final ProgrammingExerciseRepository programmingExerciseRepository;
 
+    // map of build jobs currently being processed accross all nodes
     private final IMap<Long, LocalCIBuildJobQueueItem> processingJobs;
 
-    private final FencedLock lock;
+    private int localProcessingJobs = 0;
+
+    // lock to prevent multiple nodes from processing the same build job
+    private final FencedLock fLock;
+
+    // lock for operations on single instance
+    private final ReentrantLock lock = new ReentrantLock();
```

Comment:
```suggestion
    private final FencedLock sharedLock;

    // lock for operations on single instance
    private final ReentrantLock instanceLock = new ReentrantLock();
```
Those attributes should get a name that reflects what they do rather than the current generic ones.

Additional nitpick: If you convert the comments to actual JavaDoc, the IDE probably shows it in a popup on hover over a use of the attribute.